src/prototype_room_creepbuilder.js

```diff
@@ -321,17 +316,42 @@ Room.prototype.spawnCreateCreep = function(creep) {
     buildRoad: unit.buildRoad,
```

Comment:
i supose flee and build road should be move into roles settings . As that, we can use unit.updateSettings method modify them if we want a creep with different settings than what the default ones are.